contracts/strategies/StrategyTUSDypool.sol

```diff
@@ -0,0 +1,365 @@
+/**
+ *Submitted for verification at Etherscan.io on 2020-11-16
+ */
+
+pragma solidity ^0.5.17;
+
+import "@openzeppelinV2/contracts/token/ERC20/IERC20.sol";
+import "@openzeppelinV2/contracts/math/SafeMath.sol";
+import "@openzeppelinV2/contracts/utils/Address.sol";
+import "@openzeppelinV2/contracts/token/ERC20/SafeERC20.sol";
+
+import "../../interfaces/yearn/IController.sol";
+import "../../interfaces/curve/Curve.sol";
+
+interface yvERC20 {
+    function deposit(uint256) external;
+
+    function withdraw(uint256) external;
+
+    function getPricePerFullShare() external view returns (uint256);
+}
+
+/*
+
+ A strategy must implement the following calls;
+ 
+ - deposit()
+ - withdraw(address) must exclude any tokens used in the yield - Controller role - withdraw should return to Controller
+ - withdraw(uint) - Controller | Vault role - withdraw should always return to vault
+ - withdrawAll() - Controller | Vault role - withdraw should always return to vault
+ - balanceOf()
+ 
+ Where possible, strategies must remain as immutable as possible, instead of updating variables, we update the contract by linking it in the controller
+ 
+*/
+
+contract StrategyTUSDypool {
+    using SafeERC20 for IERC20;
+    using Address for address;
+    using SafeMath for uint256;
+
+    address public constant want = address(0x0000000000085d4780B73119b644AE5ecd22b376);
+    address public constant ypool = address(0x45F783CCE6B7FF23B2ab2D70e416cdb7D6055f51);
+    address public constant ycrv = address(0xdF5e0e81Dff6FAF3A7e52BA697820c5e32D806A8);
+    address public constant yycrv = address(0x5dbcF33D8c2E976c6b560249878e6F1491Bca25c);
+
+    address public constant dai = address(0x6B175474E89094C44Da98b954EedeAC495271d0F);
+    address public constant ydai = address(0x16de59092dAE5CcF4A1E6439D611fd0653f0Bd01);
+    address public constant usdc = address(0xA0b86991c6218b36c1d19D4a2e9Eb0cE3606eB48);
+    address public constant yusdc = address(0xd6aD7a6750A7593E092a9B218d66C0A814a3436e);
+    address public constant usdt = address(0xdAC17F958D2ee523a2206206994597C13D831ec7);
+    address public constant yusdt = address(0x83f798e925BcD4017Eb265844FDDAbb448f1707D);
+    address public constant tusd = address(0x0000000000085d4780B73119b644AE5ecd22b376);
+    address public constant ytusd = address(0x73a052500105205d34Daf004eAb301916DA8190f);
+
+    address public governance;
+    address public controller;
+    address public strategist;
+    address public keeper;
+
+    uint256 public constant DENOMINATOR = 10000;
+    uint256 public treasuryFee = 500;
+    uint256 public withdrawalFee = 0;
+    uint256 public strategistReward = 50;
+    uint256 public threshold = 6000;
+    uint256 public slip = 100;
```

Comment:
To confirm, 1% slippage tolerance?

Reply:
right. the same default slippage on curve. and it's kept flexible to balance between "many calls of deposit with very small slippage" and "fewer calls of deposit with acceptable slippage".